java/core/src/main/java/sleeper/core/record/serialiser/RecordSerialiser.java

```diff
@@ -148,8 +148,12 @@ private Object read(PrimitiveType primitiveType, DataInputStream dis) throws IOE
         if (primitiveType instanceof ByteArrayType) {
             int length = dis.readInt();
             byte[] byteArray = new byte[length];
-            dis.read(byteArray);
-            return byteArray;
+            int result = dis.read(byteArray);
+            if (result == length) {
+                return byteArray;
+            } else {
+                return new byte[]{};
+            }
```

Comment:
Maybe we should throw an exception if the full byteArray wasn't read, like in CompactionJobSerDe?